### PR TITLE
tests: additional unit tests for `test_transformers.py` and fix edge cases

### LINKED ISSUE #5638: persistent_cache: This cell raised an exception: SkipWithBlock
### Describe the bug

```py
import marimo as mo
with mo.persistent_cache("foo") as c:
    try:
        bar = "baz"
    except Exception as e:
        bar = "foo"
```

The exception is raised when the cache is *loaded*.

cc @dmadisetti 

### Will you submit a PR?

- [ ] Yes

### Environment

<details>

```
Replace this line with the output of marimo env. Leave the backticks in place.
```

</details>

### Code to reproduce

_No response_

### PR DESCRIPTION
**Pull request by dmadisetti**
Added some unit testing around `marimo/_ast/transformers.py`, notably fixing / adding a check for a try block in WithSkip cases.

fixes #5638

### CODE REVIEW

File: marimo/_ast/transformers.py [hunk 1]
```diff
@@ -322,7 +336,26 @@ def generic_visit(self, node: ast.AST) -> tuple[ast.Module, ast.Module]:  # type
...
+class ExtractSkippableWithBlock(ExtractWithBlock):
+    def generic_visit(self, node: ast.AST) -> tuple[ast.Module, ast.Module]:  # type: ignore[override]
+        pre_block, with_block = super().generic_visit(node)
+        # We should fail if the first node in with_block is a try.
+        if isinstance(with_block.body[0], ast.Try):
+            raise BlockException(
+                "As a limitation of caching context, the first statement "
+                "cannot be a try block."
+                "\n"
+                "Please move the cache block inside of the try, or use a start "
```
Comment (dmadisetti) [inline marimo/_ast/transformers.py]:
```suggestion
                "Please move the cache block inside of the try, or start "
```

**Review by mscolnick — DISMISSED** [review]
